backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java

```diff
@@ -1222,27 +1224,36 @@ public void getUrlForArtifact(
                 .build();
         throw StatusProto.toStatusRuntimeException(status);
       }
-      String s3Key = null;
+      final String s3Key;
+      final String uploadId;
 
       /*Process code*/
       if (request.getArtifactType() == ArtifactType.CODE) {
         // just creating the error string
         errorMessage =
             "Code versioning artifact not found at experimentRun, experiment and project level";
         s3Key = getUrlForCode(request);
+        uploadId = null;
       } else if (request.getArtifactType() == ArtifactType.DATA) {
         errorMessage = "Data versioning artifact not found";
-        s3Key = getUrlForData(request);
+        Entry<String, String> s3KeyUploadId = getUrlForData(request);
+        s3Key = s3KeyUploadId.getKey();
+        uploadId = s3KeyUploadId.getValue();
```

Comment:
@convoliution didn't we deprecate data as artifact?

Reply:
I can't recall whether we decided to deprecate it, but maybe not:
The Client [still has the logic for uploading data as an artifact](https://github.com/VertaAI/modeldb/blob/095f2bd/client/verta/verta/client.py#L2676) (if the user doesn't use `DatasetVersion`),
and past-me didn't mark it as deprecated.